internal/formatter/formatter.go

```diff
@@ -0,0 +1,94 @@
+package formatter
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+// Format method formats files inside of the integration directory.
+func Format(packageRoot string, failFast bool) error {
+	err := filepath.Walk(packageRoot, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+		err = formatFile(path, failFast)
+		if err != nil {
+			return errors.Wrapf(err, "formatting file failed (path: %s)", path)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "walking through the integration files failed")
+	}
+	return nil
+}
+
+func formatFile(path string, failFast bool) error {
+	file := filepath.Base(path)
+	ext := filepath.Ext(file)
+
+	switch ext {
+	case ".json":
+	case ".yml":
+	default:
+		return nil
```

Comment:
Should we be returning an error here about the format/extension not being supported?

Reply:
Or, if not return an error, then log a message if we are skipping formatting of a file type?

Reply:
BTW, I just checked all integrations packages and this is the distribution of extensions, in case we want to add support for more formats in the future:

```
$ find packages -type f | awk -F\. '{print $NF}' | sort | uniq -c | sort -nr
 750 json
 561 yml
 149 hbs
  54 png
  27 md
  26 svg
   4 log
   2 jpg
   1 js
```

Reply:
The only ones in that list that look like worth supporting in the future (not this PR) are `yml.hbs` and _maybe_ `md`.

Reply:
The problem with `yml.hbs` files is that they're templates, so you can't easily assume which spaces/tabs you can adjust/remove and which ones are meaningful because of some conditional rendering.

Regarding `md` I believe it's doable, but this should be coordinated with tech writers. AFAIK they prefer to have a single web service you can use to format your docs, but our use case is a bit different (I assume we won't allow community users to use such web service).

Reply:
Makes sense to keep this as-is. We don't want to to return errors for un-formattable files or log messages either as it might get too spammy.